### PR TITLE
Semantic segment for YOLO11

### PR DESCRIPTION
**Pull request by kuazhangxiaoai**
This is a draft PR for semantic segmentation of YOLOv11, just including function and code. The doc about this PR is designing. 
I will commit my doc and pretrained weight as soon as possible.

I have read the CLA Document and I sign the CLA

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://github.com/ultralytics/actions)<sub>

**WARNING ⚠️** this PR is very large, summary may not cover all changes.

### 🌟 Summary
Adds first-class Semantic Segmentation support to Ultralytics (YOLO11) with new models, datasets, training/val/predict pipelines, and docs, alongside CI/exporter refinements and documentation updates. 🧠🧩

### 📊 Key Changes
- New Semantic Segmentation task
  - New task “semseg” wired end-to-end: trainer, validator, predictor, dataset, transforms, and format pipeline.
  - New model config: YOLO11 semantic segmentation (`yolo11-semseg.yaml`) and example dataset config (`CityEscapeYOLO.yaml`).
  - New runtime config: `semseg.yaml` (task defaults, args).
  - New data pipeline:
    - Semantic-specific mosaic, random perspective, flips, masks formatting, and verification utilities.
    - New dataset builder `build_semantic_dataset` and `SemanticDataset` with RGB-mask to one-hot conversion.
- CLI/API integration
  - Task mapping adds `semseg` to `YOLO` model: `SemSegTrainer`, `SemSegValidator`, `SemSegPredictor`.
  - Tests cover semseg training/val/predict flows.
- Exporter updates
  - CoreML: removes unsupported dynamic shapes option and enforces `batch=1`; improves NMS pipeline handling.
  - ONNX: FP16 warning limited to CPU-only ONNX case (clarified behavior).
  - NCNN: switches to exporting from TorchScript path for stability.
- CI tweaks
  - Pin `astral-sh/setup-uv` to v6 across workflows.
  - Simplify PyTest step (remove retry wrapper).
  - Dependabot auto-assigns @glenn-jocher as reviewer for dependency PRs.
- Docs improvements
  - New Semantic Segmentation docs page and nav entries.
  - Updated FLOPs/params tables for YOLO11 variants across seg/cls/pose/obb.
  - Updated images/links and minor content fixes (ONNX/CoreML args, examples).
  - Docs build no longer uses `--strict`.

### 🎯 Purpose & Impact
- Unlocks semantic segmentation with YOLO11 out of the box
  - Train/validate/predict with masks directly: ideal for scene parsing, AV, medical, and industrial use cases. 🩺🚗🏭
  - Clear example configs and dataset conversion pipeline reduce setup friction.
- More reliable packaging and deployment
  - Exporter fixes make CoreML and NCNN paths more predictable; clearer ONNX FP16 behavior on CPU.
- Smoother CI and maintenance
  - Stable uv setup, leaner PyTest runs, and auto-review improve pipeline velocity and reliability.
- Better user guidance
  - Updated metrics and docs reflect accurate model stats and consistent examples; new SemSeg docs make adoption easier.

Example: quickstart for Semantic Segmentation
```bash
# Train
yolo semseg train data=ultralytics/cfg/datasets/CityEscapeYOLO.yaml model=ultralytics/cfg/models/11/yolo11-semseg.yaml epochs=100 imgsz=512

# Validate
yolo semseg val model=path/to/best.pt

# Predict
yolo semseg predict model=path/to/best.pt source=path/to/images
```

Learn more in the Semantic Segmentation docs.

### CODE REVIEW

**Comment by UltralyticsAssistant** [comment]
👋 Hello @kuazhangxiaoai, thank you for submitting an `ultralytics/ultralytics` 🚀 PR! This is an automated response to help speed things along. An Ultralytics engineer will also review and assist you shortly. In the meantime, please review the checklist below and the PR-specific notes to ensure a smooth integration:

- ✅ **Define a Purpose**: Clearly explain the purpose of your fix or feature in your PR description, and link to any [relevant issues](https://github.com/ultralytics/ultralytics/issues). Ensure your commit messages are clear, concise, and adhere to the project's conventions.
- ✅ **Synchronize with Source**: Confirm your PR is synchronized with the `ultralytics/ultralytics` `main` branch. If it's behind, update it by clicking the 'Update branch' button or by running `git pull` and `git merge main` locally.
- ✅ **Ensure CI Checks Pass**: Verify all Ultralytics [Continuous Integration (CI)](https://docs.ultralytics.com/help/CI/) checks are passing. If any checks fail, please address the issues.
- ✅ **Update Documentation**: Update the relevant [documentation](https://docs.ultralytics.com/) for any new or modified features.
- ✅ **Add Tests**: If applicable, include or update tests to cover your changes, and confirm that all tests are passing.
- ✅ **Sign the CLA**: Please ensure you have signed our [Contributor License Agreement](https://docs.ultralytics.com/help/CLA/) if this is your first Ultralytics PR by writing "I have read the CLA Document and I sign the CLA" in a new message.
- ✅ **Minimize Changes**: Limit your changes to the **minimum** necessary for your bug fix or feature addition. _"It is not daily increase but daily decrease, hack away the unessential. The closer to the source, the less wastage there is."_ — Bruce Lee

Additional PR-specific notes to help expedite review:
- 🔄 Naming consistency: Please use “YOLO11” throughout (avoid “YOLOv11”) to align with Ultralytics branding.
- 🧩 Docs alignment: The new `docs/en/tasks/semseg.md` currently reads like instance segmentation content. Consider aligning the page title, description, examples, and images with semantic segmentation to avoid confusion.
- 🗂️ Config paths: Several new configs include absolute local paths (e.g., in `ultralytics/cfg/semseg.yaml`). Please replace with relative or example paths so CI and users can run them out-of-the-box.
- 📁 New assets/configs: Ensure new YAMLs (models/datasets) follow existing patterns and defaults used in `ultralytics/cfg/models/11` and dataset configs.
- 🧪 Tests: If new trainers/validators/predictors are added (e.g., `SemSegTrainer`, `SemSegValidator`, `SemSegPredictor`), please include unit/integration tests mirroring existing task tests to keep coverage healthy.
- 🧱 Minimal surface area: Where possible, centralize task registration via the existing `task_map` pattern and avoid broad changes outside semseg-specific files to keep the PR focused.
- 📝 Docstrings and typing: Adding concise docstrings and type hints to new public APIs will help reviewers and users.
- 🚦 CI focus: Pay special attention to style/lint and import order to keep CI green. Small formatting fixes can save review cycles. 

For more guidance, please refer to our [Contributing Guide](https://docs.ultralytics.com/help/contributing/). Don’t hesitate to leave a comment if you have any questions. Thank you for contributing to Ultralytics! 🚀🎨

**Comment by Jordan-Pierce** [comment]
@glenn-jocher in general, what changes (if any) need to be made to this PR?

**Comment by glenn-jocher** [comment]
Thanks Jordan—beyond the bot’s notes, to move this forward we need it strictly scoped to semseg: register the task only via the existing `task_map` hooks (no broad engine edits), keep any new args out of `default.yaml` and isolate them in `cfg/semseg.yaml`, replace absolute paths and normalize dataset/model YAML names, integrate outputs with the current `Results.masks`/plot pipeline rather than introducing a new results type, wire Export so ONNX/TensorRT runs for semseg with a small smoke test, add tiny-data tests that run 1 epoch for `train/val/predict/export`, and update docs to be semantic‑segmentation specific (mIoU/Dice, class palettes) and linked from the tasks index; for reference on expected patterns see the YOLO11 docs and the semantic segmentation overview at [YOLO11 docs](https://docs.ultralytics.com/models/yolo11/) and [semantic segmentation overview](https://www.ultralytics.com/glossary/semantic-segmentation).

**Comment by kuazhangxiaoai** [comment]
@glenn-jocher @Jordan-Pierce I have upload the semantic segmenta dataset(CityEcapse-YOLO-dataset) to Google Drive. Can you help me uploading the dataset to Ultralytics asset website, just like COCO-128 dataset.

**Comment by kuazhangxiaoai** [comment]
[CityEcapse-YOLO-512-small.zip](https://github.com/user-attachments/files/22788831/CityEcapse-YOLO-512-small.zip)
This is a small CityEcapse dataset for YOLO semseg task.

**Comment by glenn-jocher** [comment]
Thanks for sharing the sample—Ultralytics can’t mirror Google Drive links or PR attachments; please confirm the dataset’s redistribution license, then open a new issue or PR in `ultralytics/assets` with the zip and checksums so we can review hosting there, and add a dataset YAML that pulls from a GitHub Release using a `download:` block modeled after the COCO-Seg example in our docs; e.g.:

```yaml
# cityscapes-semseg.yaml (example)
path: cityscapes-semseg
train: images/train
val: images/val

# fill with your class map
names:
  0: road
  1: sidewalk
  # ...

download: |
  from pathlib import Path
  from ultralytics.utils.downloads import download
  dir = Path(yaml["path"])
  urls = ["https://docs.ultralytics.com/reference/utils/downloads/"]  # replace with your release asset
  download(urls, dir=dir.parent)
```

For the YAML structure and `download:` pattern, see the COCO-Seg dataset YAML example in the docs. [COCO‑Seg dataset YAML example](https://docs.ultralytics.com/datasets/segment/coco/)
